### DISCUSSION TITLE
Unable to proceed to login to singpass

### DISCUSSION THREAD

**Discussion by casuarinao8** [singpass/demo-app#19] [Q&A]
Hi all,

I am facing and issue when trying to test my connection for singpass myinfo, whenever it brings me to the singpass page, it shows me the following error. Could you please advise what is the issue? The client ID and staging application has already been activated. 
<img width="410" alt="Singpass Issue Integration" src="https://github.com/user-attachments/assets/a00ae562-4024-4b7e-99be-43f2efccc5cb">
ef">

**Comment by luiscvega** [comment]
HI @casuarinao8. Can you share the the `auth` URL that you used? You also to need to make sure that the Redirect URI matches that with your Staging settings, and that you have a valid accessible JWKS URL or JWKS Object in the Staging settings. You can use the JSON from the demo-app for now if you'd like:

```json
{
  "keys": [
    {
      "alg": "ES256",
      "kty": "EC",
      "x": "tqG7PiAPD0xTBKdxDd4t8xAjJleP3Szw1CZiBjogmoc",
      "y": "256TjvubWV-x-C8lptl7eSbMa7pQUXH9LY1AIHUGINk",
      "crv": "P-256",
      "use": "sig",
      "kid": "my-sig-key"
    },
    {
      "alg": "ECDH-ES+A256KW",
      "kty": "EC",
      "x": "_TSrfW3arG1Ebc8pCyT-r5lAFvCh_rJvC5HD5-y8yvs",
      "y": "Sr2vpuU6gzdUiXddGnRJIroXCfdameaR1mgU49H5h9A",
      "crv": "P-256",
      "use": "enc",
      "kid": "my-enc-key"
    }
  ]
}
```

**Reply by casuarinao8** [reply]
May I know how do I get the private key for this demo app?

**Reply by luiscvega** [reply]
It's in the [config.json file](https://github.com/singpass/demo-app/blob/main/examples/config.json) in the `demo-app/examples` folder. Here is the contents of the file:

```json
{
    "SERVER_PORT": 3080,
    "CLIENT_ID": "RsrOy2iB0edR53TJSuD5ULad1pGmrVZL",
    "ISSUER_URL": "https://stg-id.singpass.gov.sg",
    "REDIRECT_URI": "http://localhost:3080/callback",
    "SCOPES": "openid uinfin name",
    "KEYS": {
        "PRIVATE_SIG_KEY": {
          "alg": "ES256",
          "kty": "EC",
          "x": "tqG7PiAPD0xTBKdxDd4t8xAjJleP3Szw1CZiBjogmoc",
          "y": "256TjvubWV-x-C8lptl7eSbMa7pQUXH9LY1AIHUGINk",
          "crv": "P-256",
          "d": "PgL1UKVpvg_GeKdxV-oUEPIDhGBP2YYZLGiZ5HXDZDI",
          "kid": "my-sig-key"
        },
        "PRIVATE_ENC_KEY": {
          "alg": "ECDH-ES+A256KW",
          "kty": "EC",
          "x": "_TSrfW3arG1Ebc8pCyT-r5lAFvCh_rJvC5HD5-y8yvs",
          "y": "Sr2vpuU6gzdUiXddGnRJIroXCfdameaR1mgU49H5h9A",
          "crv": "P-256",
          "d": "AEabUwi3VjOOfiyoOtSGrqpl8cfhcUhNtj-xh1l-UYE",
          "kid": "my-enc-key"
        },
        "PUBLIC_SIG_KEY": {
          "alg": "ES256",
          "kty": "EC",
          "x": "tqG7PiAPD0xTBKdxDd4t8xAjJleP3Szw1CZiBjogmoc",
          "y": "256TjvubWV-x-C8lptl7eSbMa7pQUXH9LY1AIHUGINk",
          "crv": "P-256",
          "use": "sig",
          "kid": "my-sig-key"
        },
        "PUBLIC_ENC_KEY": {
          "alg": "ECDH-ES+A256KW",
          "kty": "EC",
          "x": "_TSrfW3arG1Ebc8pCyT-r5lAFvCh_rJvC5HD5-y8yvs",
          "y": "Sr2vpuU6gzdUiXddGnRJIroXCfdameaR1mgU49H5h9A",
          "crv": "P-256",
          "use": "enc",
          "kid": "my-enc-key"
        }
      }
}
```

**Comment by dannycsz** [comment]
I use the exact same demo code except changing the client ID after creating the staging app, putting the demo public key JWKS Object since the portal doesn't allow to input http://localhost:3080/ without https. 

I still face this error

<img width="1275" alt="image" src="https://github.com/user-attachments/assets/0b24d892-1b55-4238-9c01-682a37c13072">
